rfcs/convergence/changelog-guidelines.md

```diff
@@ -0,0 +1,130 @@
+the PR. This makes the change harder to understand since the contents might only apply vaguely, or not at all to the package.
+
+A simple example would be `Fix dependencies`, this kind of generic description can be applied to a PR when linked to
+an issue, but has an incredibly vague meaning with applied to the changelog of a package.
+
+## Detailed design or proposal
+
+### Semantic labels for changefile entries
+
+Each changefile entry should have a message that has a semantic label that is inspired by [conventional commits](https://www.conventionalcommits.org/en/v1.0.0/).
+This RFC proposes four labels in the following form:
+
+```
+feat:  This label communicates that a feature was added to the package
+fix:   This label communicates that a fix was applied to the package
+chore: This label communicates that changes which should not have an affect on th user were applied
+docs:  This label communicates that documentation was changed
+```
+
+We can configure [danger.js](https://danger.systems/js/) to validate changefile entries in the PR that a correct
+label has been applied.
+
+### Stricter reviews for changefile comments
+
+There is no easy way to automate the quality of a changelog entry, this has to be done by humans in the PR review.
+This proposal tries to spread awareness that changefile entries are directly a part of our changelog and they should
+be reviewed strictly.
+
+### Stricter review of `type` and `dependentChangeType`
+
+Here is a sample beachball changefile:
+
+```json
+{
+  "type": "patch",
+  "comment": "This comment is the message used in the changelog entry",
+  "packageName": "@fluentui/babel-make-styles",
+  "email": "jane.doe@contoso.com",
+  "dependentChangeType": "patch"
+}
+```
+
+The value of `type` determines what kind of version bump will be applied to the package on release. It can take the
+following values:
+
+- none
+- prerelease
+- patch
+- minor
+- major
+
+These values are identical to standard semver bumps, and will result in bumps of that type being applied to the package
+on release. Beachball will always pick the highest bump type among all changefiles for a package when bumping. A value
+of `none` will result in no bumps applied to the package. The `comment` will not be applied to the changelog when `none`
+is used for `type`.
+
+The value of `dependentChangeType` is the same as `type`. However, this bump affects upstream packages that depend on the
+package that the changefile references. Each upstream package will also have an added changelog entry which resembles
+
+> Bump @fluentui/react-aria to v9.0.0-beta.4 ([PR #20762](https://github.com/microsoft/fluentui/pull/20762) by beachball)
+
+These kinds of 'bump' changelog entries are omitted from `react-components` when rolled up into the final changelog.
+
+It could be possible to automatically enforce incorrect values for `type` and `dependentType` to some extent by
```

Comment:
The only times we'd need to modify `dependentChangeType` are:
- Modifications to an API which is re-exported by another package => could possibly detect from api.md changes
- Modification to devDependency (eslint-plugin, react-conformance) => we should update beachball to allow automatically setting `dependentChangeType: none` for such packages

Reply:
> we should update beachball to allow automatically setting dependentChangeType: none for such packages

Not necessarily right ? since if we update something like `babel-make-styles` it's a dev dependency change but could actually impact our shipped code (styles)

Reply:
It would be opt-in per package, only used for things like `eslint-plugin` or `react-conformance` that definitely shouldn't cause dependent bumps.